### PR TITLE
Resolved error in article and category page and added test cases.

### LINKED ISSUE #207: Articles page is broken on staging
@sudeeptarlekar _a

### PR DESCRIPTION
**Pull request by sudeeptarlekar**
This fixes #207 

@breezyboa _a Please review

CC: @vipulnsward

### CODE REVIEW

File: app/javascript/Client/Data/Category.elm [hunk 1]
```diff
@@ -36,7 +43,7 @@ allCategoriesQuery =
                     (GQLBuilder.object Category
                         |> GQLBuilder.with (GQLBuilder.field "id" [] GQLBuilder.string)
                         |> GQLBuilder.with (GQLBuilder.field "name" [] GQLBuilder.string)
-                        |> GQLBuilder.with articleSummaryField
+                        |> GQLBuilder.with (GQLBuilder.field "articles" [] (GQLBuilder.list articlesExtractor))
                     )
```
Comment (breezyboa) [inline app/javascript/Client/Data/Category.elm]: Revert this change.
```
|> GQLBuilder.with (GQLBuilder.field "articles" [] (GQLBuilder.list articlesExtractor))
```
will cause a nested field error

**Comment by breezyboa** [comment]
@sudeeptarlekar _a

**Comment by sudeeptarlekar** [comment]
> Revert this change.

Without this change categories graphql query is breaking with error `Field 'articles' doesn't exist on type 'Article'`

**Comment by sudeeptarlekar** [comment]
@breezyboa _a reverted change as requested.

**Comment by breezyboa** [comment]
@vipulnsward _a can you review rails code

**Pull request merged**